src/transformers/models/vilt/modeling_vilt.py

```diff
@@ -811,12 +811,16 @@ def forward(
         if attention_mask is None:
             attention_mask = torch.ones(((batch_size, seq_length)), device=device)
 
-        if pixel_values is None:
-            raise ValueError("You have to specify pixel_values")
+        if pixel_values is not None and image_embeds is not None:
+            raise ValueError("You cannot specify both pixel_values and image_embeds at the same time")
+        elif pixel_values is None and image_embeds is None:
+            raise ValueError("You have to specify either pixel_values or image_embeds")
 
-        batch_size, num_channels, height, width = pixel_values.shape
+        image_batch_size = pixel_values.shape[0] if pixel_values is not None else image_embeds.shape[0]
+        if image_batch_size != batch_size:
+            raise ValueError("The text inputs and image inputs need to have the same batch size")
```

Comment:
Maybe for consistency, we can call the other one `text_batch_size` (instead of `batch_size`).

Reply:
done :-) & wait for green CI